userland/src/parameters.cpp

```diff
@@ -0,0 +1,74 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+template <typename ParameterTypeT>
+ParameterTypeT
+wait_for_future(rclcpp::node::Node::SharedPtr & node, std::shared_future<ParameterTypeT> & future)
+{
+  std::future_status status;
+  do {
+    rclcpp::spin_some(node);
+    status = future.wait_for(std::chrono::milliseconds(100));
+  } while (status != std::future_status::ready && rclcpp::ok());
+
+  if (std::future_status::ready == status) {
+    return future.get();
+  }
+
+  throw std::runtime_error("Invalid future status");
+}
+
+int main(int argc, char** argv)
+{
+  rclcpp::init(argc, argv);
+
+  const std::string node_name("paramers_example");
+
+  rclcpp::node::Node::SharedPtr node = rclcpp::Node::make_shared(node_name);
```

Comment:
I think we should try to keep the examples concise, I would have done this:

```c++
auto node = rclcpp::Node::make_shared("parameters_example");
```

Why expand it out with a separate variable for the node name? I guess you just want to be explicit?